### PR TITLE
Add a command to remove older trust

### PR DESCRIPTION
**Pull request by stlaz**
Adds the `oc adm ocp-certificates remove-old-trust` command.

TODO:
- [x] add an easy way to exclude certain CMs
- [x] add unit tests
- [ ] test to see if it works

/assign @deads2k 

built on top of https://github.com/openshift/oc/pull/1443

### CODE REVIEW

File: pkg/cli/admin/ocpcertificates/trustpurge/trustpurge.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	}
+
+	if r.dryRun {
+		applyOptions.DryRun = []string{metav1.DryRunAll}
+	}
+
+	applyCM := applycorev1.ConfigMap(cm.Name, cm.Namespace)
+	applyCM = applyCM.WithData(newCMData)
+
+	finalObj, err := r.KubeClient.CoreV1().ConfigMaps(cm.Namespace).Apply(context.TODO(), applyCM, applyOptions)
```
[the remark this answers is not in the document]
Comment (stlaz) [reply pkg/cli/admin/ocpcertificates/trustpurge/trustpurge.go:126]: Looks like resourceversion constraint is only available for `Get`, `List` and `Delete` actions

**Comment by deads2k** [comment]
followup we should attempt.  read all the secrets that hold the targets and see if the CA bundle we would write can validate them.  That would protect us in case someone runs this command inappropriately.

File: pkg/cli/admin/ocpcertificates/certregen/regenerate.go [hunk 2]
```diff
@@ -61,10 +61,6 @@ func (o *RegenerateCertsRuntime) forceRegenerationOnResourceInfo(info *resource.
 		return err
 	}
 
-	if secretKind != info.Object.GetObjectKind().GroupVersionKind() {
```
Comment (deads2k) [inline pkg/cli/admin/ocpcertificates/certregen/regenerate.go:64]: please leave in place.

File: pkg/cli/admin/ocpcertificates/certregen/rootregen_test.go [hunk 3]
```diff
@@ -114,7 +128,7 @@ func testPrinter(runtime.Object, io.Writer) error {
 }
 
 func testSecret(t *testing.T) *corev1.Secret {
-	key, err := rsa.GenerateKey(rand.Reader, 4096)
+	key, err := rsa.GenerateKey(rand.Reader, 1024)
```
Comment (deads2k) [inline pkg/cli/admin/ocpcertificates/certregen/rootregen_test.go:131]: 2048 or fips gets angry.  Yes, even in tests.

File: pkg/cli/admin/ocpcertificates/trustpurge/trustpurge.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+func (r *RemoveOldTrustRuntime) purgeTrustFromConfigMap(cm *corev1.ConfigMap) error {
+	var err error
+	var finalObj *corev1.ConfigMap
+
+	if excludedConfigMaps := r.excludeBundles[cm.Namespace]; excludedConfigMaps.Has(cm.Name) {
+		return nil
+	}
+
+	for retriesLeft := 2; retriesLeft > 0; retriesLeft -= 1 {
```
Comment (deads2k) [inline pkg/cli/admin/ocpcertificates/trustpurge/trustpurge.go:86]: Use the retryonconflict method

**Comment by deads2k** [comment]
solid starting point.  We should write a precondition check that the target cert involved can be validated using the keys that will remain.  That will likely end up *requiring* the date.

/lgtm
/approve

**Pull request merged**